src/models/users.js

```diff
@@ -0,0 +1,44 @@
+const mongoose = require("mongoose");
+Schema = mongoose.Schema;
+
+const userSchema = new Schema({
+    user_id: {
+        type: Schema.Types.ObjectId,
+        required: true,
+        unique: true,
+        default: mongoose.Types.ObjectId // Generate a new ObjectId by default
+    },
+    userName: {
+        type: String,
+        required: true,
+        index: { unique: true } // Create an index for faster lookup and enforce uniqueness
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    password: {
+        type: String,
+        required: true,
+    },
+    address: { 
+        type : Array, 
```

Comment:
instade of Array why not [String]

Reply:
what i am thinking we should be precise about it and add an array of objects which contains city, street, number, etc. THIS IS A NOTE will be discussed in the meeting before the last arragements.

Reply:
A question:  is it necessary to include required: true for the user_id field, as the default value generation and unique constraint ensure its presence in every document?